public/app/plugins/datasource/grafana-azure-monitor-datasource/ConfigEditor.tsx

```diff
@@ -0,0 +1,304 @@
+      await this.getWorkspaces();
+    }
+  };
+
+  getLogAnalyticsSubscriptions = async () => {
+    if (!this.logAnalyticsHasNecessaryCredentials()) {
+      return;
+    }
+
+    const azureMonitorDatasource = new AzureMonitorDatasource(this.state.config, this.backendSrv, this.templateSrv);
+
+    let logAnalyticsSubscriptions = (await azureMonitorDatasource.getSubscriptions('workspacesloganalytics')) || [];
+    logAnalyticsSubscriptions = logAnalyticsSubscriptions.map(subscription => {
+      return {
+        value: subscription.value,
+        label: subscription.text,
+      };
+    });
+
+    if (logAnalyticsSubscriptions && logAnalyticsSubscriptions.length > 0) {
+      this.setState({ logAnalyticsSubscriptions });
+
+      this.state.config.editorJsonData.logAnalyticsSubscriptionId =
+        this.state.config.editorJsonData.logAnalyticsSubscriptionId || logAnalyticsSubscriptions[0].value;
+    }
+
+    if (this.state.config.editorJsonData.logAnalyticsSubscriptionId) {
+      await this.getWorkspaces();
+    }
+  };
+
+  getWorkspaces = async () => {
+    const sameAs =
+      this.state.config.jsonData.azureLogAnalyticsSameAs && this.state.config.editorJsonData.subscriptionId;
+    if (!sameAs && !this.state.config.editorJsonData.logAnalyticsSubscriptionId) {
+      return;
+    }
+
+    const azureLogAnalyticsDatasource = new AzureLogAnalyticsDatasource(
+      this.state.config,
+      this.backendSrv,
+      this.templateSrv
+    );
+
+    let logAnalyticsWorkspaces = await azureLogAnalyticsDatasource.getWorkspaces(
+      sameAs
+        ? this.state.config.editorJsonData.subscriptionId
+        : this.state.config.editorJsonData.logAnalyticsSubscriptionId
+    );
+    logAnalyticsWorkspaces = logAnalyticsWorkspaces.map(workspace => {
+      return {
+        value: workspace.value,
+        label: workspace.text,
+      };
+    });
+
+    if (logAnalyticsWorkspaces.length > 0) {
+      this.setState({ logAnalyticsWorkspaces });
+
+      this.state.config.editorJsonData.logAnalyticsDefaultWorkspace =
+        this.state.config.editorJsonData.logAnalyticsDefaultWorkspace || logAnalyticsWorkspaces[0].value;
+    }
+  };
+
+  render() {
+    const { config, subscriptions, logAnalyticsSubscriptions, logAnalyticsWorkspaces } = this.state;
+
+    return (
+      <>
+        <MonitorConfig
+          datasourceConfig={config}
+          subscriptions={subscriptions}
+          onLoadSubscriptions={this.onLoadSubscriptions}
+          onDatasourceUpdate={this.onConfigUpdate}
+        />
+
+        <AnalyticsConfig
+          datasourceConfig={config}
+          logAnalyticsWorkspaces={logAnalyticsWorkspaces}
+          logAnalyticsSubscriptions={logAnalyticsSubscriptions}
+          onLoadSubscriptions={this.onLoadSubscriptions}
+          onDatasourceUpdate={this.onConfigUpdate}
+          onLoadWorkspaces={this.getWorkspaces}
+        />
+
```

Comment:
when the "same details as Azure Monitor API is not checked, there is a stray "0" below "Default Workspace".

When you fill in the Application (client) ID it goes away.

![image](https://user-images.githubusercontent.com/7364245/62406433-657af680-b571-11e9-919c-615da5505917.png)

with client id:

![image](https://user-images.githubusercontent.com/7364245/62406445-8cd1c380-b571-11e9-8eac-caa7d67ba237.png)

Reply:
The "Load Subscriptions" should be displayed (greyed out) when the fields are not ready, then turn Blue when ready.

![image](https://user-images.githubusercontent.com/7364245/62406463-f81b9580-b571-11e9-8aed-4a0dab3a29fd.png)

Reply:
Subscriptions dropdown should also be a filtering/typable field where the user can paste in the info, or filter results by typing (there may be hundreds of subscriptions)

Workspaces should also operate the same way (there may be hundreds of workspaces, locating them by scrolling will be tedious, and we may hit a limit on how many get displayed).

We should add an icon/tooltip that indicates you can paste/type in these fields.

Reply:
Also - is there a limit to how many subscriptions are loaded?

Reply:
We need to add a tooltip to "Same details as Azure Monitor API" if it is checked, it should say
"Workspaces are pulled from default subscription selected above"

![image](https://user-images.githubusercontent.com/7364245/62406539-34032a80-b573-11e9-83ef-cbcedcff4cbb.png)

Reply:
Changed the buttons to disabled instead of hiding per other comments so the 0 doesn't show anymore now anyway.

The Select field is searchable by default. You just have to start typing- this is both workspaces and subscriptions. Copy for copy/paste?

There's no limit to my knowledge.

Added tooltip to the switch component so now we have tooltips on switches :). That tip has been added.

Reply:
All looks good to me.